src/Umbraco.Web/UmbracoApplication.cs

```diff
@@ -17,16 +18,24 @@ protected override IRuntime GetRuntime()
         {
             var logger = SerilogLogger.CreateWithDefaultConfiguration();
 
+            var runtime = new WebRuntime(this, logger, GetMainDom(logger));
+
+            return runtime;
+        }
+
+        /// <summary>
+        /// Returns a new MainDom
+        /// </summary>
+        public static IMainDom GetMainDom(ILogger logger)
```

Comment:
probably should be protected (not static). This allows for more flexibility. I'll update